pkg/deploy/deploy_rp.go

```diff
@@ -186,3 +192,23 @@ func (d *deployer) convertToIPAddressOrRange(ipSlice []string) []mgmtdocumentdb.
 	}
 	return ips
 }
+
+func (d *deployer) rpRemoveFailedScaleset(ctx context.Context) error {
+	d.log.Print("removing failed scaleset")
+
+	// Name is [concat('rp-vmss-', parameters('vmssName'))]; vmssName is d.version
+	vmss, err := d.vmss.Get(ctx, d.config.RPResourceGroupName, "rp-vmss-"+d.version)
+	if err != nil {
+		return err
+	}
+
+	if *vmss.ProvisioningState == "Failed" {
```

Comment:
Use constant `ProvisioningStateFailed` from azure sdk instead of hard coded string.

Reply:
Fixed